### PR TITLE
[ASL] Use Clang for FreeBSD, some other fixes

### PR DESCRIPTION
**Pull request by giordano**
Ref https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/512.  CC: @dpo

### CODE REVIEW

File: A/ASL/build_tarballs.jl [hunk 1]
```diff
@@ -26,36 +25,26 @@ cflags=""
...
+elif [[ "${target}" == "aarch64-linux-musl" ]]; then
     cp $WORKSPACE/srcdir/asl-extra/arith.h.aarch64-linux-gnu ./arith.h
-fi
-if [ $target = "x86_64-unknown-freebsd11.1" ]; then
+elif [[ "${target}" == *-freebsd* ]]; then
     cp $WORKSPACE/srcdir/asl-extra/arith.h.x86_64-linux-gnu ./arith.h
     cflags="-D__XSI_VISIBLE=1"
-fi
-
-if [ $target = "x86_64-w64-mingw32" || $target = "i686-w64-mingw32" ]; then
```
Comment (giordano) [inline A/ASL/build_tarballs.jl]: This syntax is wrong:
```
sandbox:${WORKSPACE} # [ $target = "x86_64-w64-mingw32" || $target = "i686-w64-mingw32" ]
bash: [: missing `]'
bash: x86_64-unknown-freebsd11.1: command not found
```
I'm surprised that this didn't make builders to fail

Comment (dpo) [reply A/ASL/build_tarballs.jl]: Gee, you're right. It never complained locally either. Many thanks for looking into this!

Comment (giordano) [reply A/ASL/build_tarballs.jl]: Actually in the log I could see the error, but that didn't make the build to fail, which is strange :confused:

Comment (dpo) [reply A/ASL/build_tarballs.jl]: At any rate, I confirm that your changes also work for me locally.

**Comment by dpo** [comment]
Is there a way to test this before it is merged?

**Comment by giordano** [comment]
Test what exactly?

**Comment by dpo** [comment]
Test the Julia package that uses these binaries.

**Comment by giordano** [comment]
Not using Yggdrasil, because the tarballs are uploaded and the JLL package is registered only for commits on `master` (including merged pull requests), not for pull requests.

We don't want to generated a release for each build triggered by pull requests, they're usually _a lot_.

If you really want to test, you can build locally, upload the tarballs somewhere and use them for testing.  This is what I used to do before Yggdrasil anyway.

If you're concerned about the FreeBSD binaries, using GCC was for sure an issue.  As I wrote in https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/512#issuecomment-557816665, the library was looking for a GCC library that was nowhere on the system.  I verified that by using Clang the library is linked only to existing system libraries.

**Comment by dpo** [comment]
Ok, thanks.

**Comment by giordano** [comment]
I'm waiting for https://github.com/JuliaRegistries/General/pull/5667 to be merged before merging this

**Pull request merged**